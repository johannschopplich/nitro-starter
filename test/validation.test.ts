import { describe, expect, it } from 'vitest'
import { $fetch } from 'nitro-test-utils'

describe('validate requests using valibot schemas', () => {
  describe('validate query', () => {
    it('returns 200 if query matches validation schema', async () => {
      const { body, status } = await $fetch('/api/validate/query?user=nitro')

      expect(status).toBe(200)
      expect(body).toMatchSnapshot()
    })

    it('throws 400 if query does not match validation schema', async () => {
      const { body, status } = await $fetch('/api/validate/query')

      expect(status).toBe(400)
      expect(body).toMatchSnapshot()
    })
  })

  describe('validate body', () => {
    it('returns 200 if body matches validation schema', async () => {
      const { body, status } = await $fetch('/api/validate/body', {
        method: 'POST',
        body: JSON.stringify({ required: true }),
        headers: {
          'content-type': 'application/json',
        },
      })

      expect(status).toBe(200)
      expect(body).toMatchSnapshot()
    })

    it('throws 400 if body does not match validation schema', async () => {
      const { body, status } = await $fetch('/api/validate/body', {
        method: 'POST',
        body: JSON.stringify({}),
        headers: {
          'content-type': 'application/json',
        },
      })

      expect(status).toBe(400)
      expect(body).toMatchSnapshot()
    })
  })
})
