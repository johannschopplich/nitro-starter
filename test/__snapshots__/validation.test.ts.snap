// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`validate requests using valibot schemas > validate body > returns 200 if body matches validation schema 1`] = `
{
  "required": true,
}
`;

exports[`validate requests using valibot schemas > validate body > throws 400 if body does not match validation schema 1`] = `
{
  "message": "Invalid type: Expected boolean but received undefined",
  "statusCode": 400,
  "statusMessage": "Invalid type: Expected boolean but received undefined",
  "url": "/api/validate/body",
}
`;

exports[`validate requests using valibot schemas > validate query > returns 200 if query matches validation schema 1`] = `
{
  "user": "nitro",
}
`;

exports[`validate requests using valibot schemas > validate query > throws 400 if query does not match validation schema 1`] = `
{
  "message": "Invalid type: Expected string but received undefined",
  "statusCode": 400,
  "statusMessage": "Invalid type: Expected string but received undefined",
  "url": "/api/validate/query",
}
`;
